crates/resolver-tests/src/helpers.rs

```diff
@@ -0,0 +1,247 @@
+    fn to_dep(self) -> Dependency {
+        self
+    }
+    fn opt(mut self) -> Dependency {
+        self.set_optional(true);
+        self
+    }
+    fn with(mut self, features: &[&'static str]) -> Dependency {
+        self.set_default_features(false);
+        self.set_features(features.into_iter().copied());
+        self
+    }
+    fn with_default(mut self) -> Dependency {
+        self.set_default_features(true);
+        self
+    }
+    fn rename(mut self, name: &str) -> Dependency {
+        self.set_explicit_name_in_toml(name);
+        self
+    }
+}
+
+pub trait ToPkgId {
+    fn to_pkgid(&self) -> PackageId;
+}
+
+impl ToPkgId for PackageId {
+    fn to_pkgid(&self) -> PackageId {
+        *self
+    }
+}
+
+impl<'a> ToPkgId for &'a str {
+    fn to_pkgid(&self) -> PackageId {
+        PackageId::try_new(*self, "1.0.0", registry_loc()).unwrap()
+    }
+}
+
+impl<T: AsRef<str>, U: AsRef<str>> ToPkgId for (T, U) {
+    fn to_pkgid(&self) -> PackageId {
+        let (name, vers) = self;
+        PackageId::try_new(name.as_ref(), vers.as_ref(), registry_loc()).unwrap()
+    }
+}
+
+#[macro_export]
+macro_rules! pkg {
+    ($pkgid:expr => [$($deps:expr),* $(,)? ]) => ({
+        use $crate::helpers::ToDep;
+        let d: Vec<Dependency> = vec![$($deps.to_dep()),*];
+        $crate::helpers::pkg_dep($pkgid, d)
+    });
+
+    ($pkgid:expr) => ({
+        $crate::helpers::pkg($pkgid)
+    })
+}
+
+fn registry_loc() -> SourceId {
+    static EXAMPLE_DOT_COM: OnceLock<SourceId> = OnceLock::new();
+    let example_dot = EXAMPLE_DOT_COM.get_or_init(|| {
+        SourceId::for_registry(&"https://example.com".into_url().unwrap()).unwrap()
+    });
+    *example_dot
+}
+
+pub fn pkg<T: ToPkgId>(name: T) -> Summary {
+    pkg_dep(name, Vec::new())
+}
+
+pub fn pkg_dep<T: ToPkgId>(name: T, dep: Vec<Dependency>) -> Summary {
+    let pkgid = name.to_pkgid();
+    let link = if pkgid.name().ends_with("-sys") {
+        Some(pkgid.name())
+    } else {
+        None
+    };
+    Summary::new(name.to_pkgid(), dep, &BTreeMap::new(), link, None).unwrap()
+}
+
+pub fn pkg_dep_with<T: ToPkgId>(
+    name: T,
+    dep: Vec<Dependency>,
+    features: &[(&'static str, &[&'static str])],
+) -> Summary {
+    let pkgid = name.to_pkgid();
+    let link = if pkgid.name().ends_with("-sys") {
+        Some(pkgid.name())
+    } else {
+        None
+    };
+    let features = features
+        .into_iter()
+        .map(|&(name, values)| (name.into(), values.into_iter().map(|&v| v.into()).collect()))
+        .collect();
+    Summary::new(name.to_pkgid(), dep, &features, link, None).unwrap()
+}
+
+pub fn pkg_dep_link<T: ToPkgId>(name: T, link: &str, dep: Vec<Dependency>) -> Summary {
+    Summary::new(name.to_pkgid(), dep, &BTreeMap::new(), Some(link), None).unwrap()
+}
+
+pub fn local_pkg_id(name: &str) -> PackageId {
+    PackageId::try_new(
+        name,
+        "1.0.0",
+        SourceId::for_path(Path::new("/some-local-path")).unwrap(),
```

Comment:
In pubgrub-crates-benchmark I needed to do something a little more complicated to make it work on Windows. https://github.com/Eh2406/pubgrub-crates-benchmark/blob/main/src/cargo_resolver.rs#L131 

In retrospect I wonder if it would be easier to use `SourceId::for_registry(&"https://other.example.com".into_url().unwrap()).unwrap()` in both projects.